rmw_zenoh_cpp/src/detail/liveliness_utils.hpp

```diff
@@ -227,6 +230,35 @@ std::optional<rmw_qos_profile_t> keyexpr_to_qos(const std::string & keyexpr);
 ///=============================================================================
 /// Convert a Zenoh id to a string.
 std::string zid_to_str(const z_id_t & id);
+
+///=============================================================================
+// Helper function to convert string to size_t.
+// The function is templated to enable conversion to size_t or std::size_t.
+template<typename T>
+std::optional<T> str_to_size_t(const std::string & str, const T default_value)
+{
+  if (str.empty()) {
+    return default_value;
+  }
+  errno = 0;
+  char * endptr;
+  size_t num = strtoul(str.c_str(), &endptr, 10);
```

Comment:
Thinking about it more today, I realized two problems here:

1.  `strtoul` is documented as returning an `unsigned long`, not a `size_t`.  This is an existing bug.
2.  If T ended up being a signed integer, we'd likely do the incorrect thing here.  We could fix this either by making the template type only support unsigned (any signed integers would fail to compile), or we could do some other template magic to choose `strtoul` or `strtol` as appropriate.

I would say neither of those are blockers, but if you decide not to fix them I'd appreciate a TODO comment explaining what should be improved here.